### PR TITLE
Don't add bonus damage to pct health.

### PR DESCRIPTION
**Pull request by SeTM**

### CODE REVIEW

**Comment by Shauren** [comment]
Some explanation would be nice why this is a fix and not a random change you think it should work like.

**Comment by SeTM** [comment]
If you look at the complete code, you should be all clear. Below is a count on how many health damage. (damage = uint32(target->CountPctFromMaxHealth(damage)); If we consider the bonus damage instead of 10% we can get 319% of health for example.

**Comment by Frytiks** [comment]
Can this fix problem, why etc gargoyle strike doing 2 time more damage than should ?

**Comment by SeTM** [comment]
No, it's for SPELL_AURA_PERIODIC_DAMAGE_PERCENT

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp [hunk 1]
```diff
@@ -5834,13 +5834,15 @@ void AuraEffect::HandlePeriodicDamageAurasTick(Unit* target, Unit* caster) const
     // AOE spells are not affected by the new periodic system.
     bool isAreaAura = m_spellInfo->Effects[m_effIndex].IsAreaAuraEffect() || m_spellInfo->Effects[m_effIndex].IsEffect(SPELL_EFFECT_PERSISTENT_AREA_AURA);
     // ignore negative values (can be result apply spellmods to aura damage
-    uint32 damage = std::max(GetAmount() + GetBonusAmount(), 0); // if isAreaAura == true, GetBonusAmount == 0.
+    uint32 damage = std::max(GetAmount(), 0);
 
     // Script Hook For HandlePeriodicDamageAurasTick -- Allow scripts to change the Damage pre class mitigation calculations
     sScriptMgr->ModifyPeriodicDamageAurasTick(target, caster, damage);
```
Comment (jackpoz) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp]: is this totally ignored with this PR change ?

Comment (SeTM) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp]: Yes.

**Comment by Carbenium** [comment]
Is this PR ready for merge or any objections?

**Comment by jackpoz** [comment]
@Carbenium it breaks https://github.com/TrinityCore/TrinityCore/pull/14302#discussion-diff-26026531R5840 hook as described above, up to the merger to decide if that's ok or not

**Comment by jackpoz** [comment]
@SeTM can you provide details about how to reproduce the issue and how the behavior changes after applying this PR so that it can be tested ?

**Comment by Aokromes** [comment]
Any new on this @SeTM

**Comment by jackpoz** [comment]
This PR is still lacking clear How To Reproduce steps, it also breaks an existing hook in an undocumented way. It would be nice if the hook functionality could be preserved.

**Pull request closed without merging**